Kinvey.Core/Store/DataStore.cs

```diff
@@ -277,9 +277,20 @@ public static DataStore<T> Collection(string collectionName, DataStoreType type,
 				listIDs.Add(entityID);
 			}
 
-			FindRequest<T> findByQueryRequest = new FindRequest<T>(client, collectionName, cache, storeType.ReadPolicy, DeltaSetFetchingEnabled, cacheResults, null, listIDs);
+			var cacheDelegate = new KinveyDelegate<List<T>>
+			{
+				onSuccess = (listCacheResults) => {
+					cacheResult.onSuccess(listCacheResults.FirstOrDefault());
+				},
+				onError = (error) => {
+					cacheResult.onError(error);
+				}
+			};
+
+			FindRequest<T> findByQueryRequest = new FindRequest<T>(client, collectionName, cache, storeType.ReadPolicy, DeltaSetFetchingEnabled, cacheDelegate, null, listIDs);
 			ct.ThrowIfCancellationRequested();
-			return await findByQueryRequest.ExecuteAsync();
+			var results = await findByQueryRequest.ExecuteAsync();
+			return results.FirstOrDefault();
```

Comment:
@tejasranade Good question.  For empty results, returning `FirstOrDefault` effectively does that, because an empty list will return a default object (i.e. null).  And for a `null` list, I believe that an exception will be thrown earlier, as this would be the result of a failed deserialization of the response.  But I can add a null check here to be safe.